src/App/Pages/Accounts/SetPasswordPageViewModel.cs

```diff
@@ -183,24 +198,27 @@ public async Task SubmitAsync()
                 KdfMemory = kdfConfig.Memory,
                 KdfParallelism = kdfConfig.Parallelism,
                 OrgIdentifier = OrgIdentifier,
-                Keys = new KeysRequest
-                {
-                    PublicKey = newPublicKey,
-                    EncryptedPrivateKey = newProtectedPrivateKey.EncryptedString
-                }
+                Keys = keysRequest
             };
 
             try
             {
-                await _deviceActionService.ShowLoadingAsync(AppResources.CreatingAccount);
+                await _deviceActionService.ShowLoadingAsync(AppResources.Loading);
                 // Set Password and relevant information
                 await _apiService.SetPasswordAsync(request);
                 await _stateService.SetKdfConfigurationAsync(kdfConfig);
                 await _cryptoService.SetUserKeyAsync(newUserKey);
                 await _cryptoService.SetMasterKeyAsync(newMasterKey);
                 await _cryptoService.SetMasterKeyHashAsync(localMasterPasswordHash);
                 await _cryptoService.SetMasterKeyEncryptedUserKeyAsync(newProtectedUserKey.EncryptedString);
-                await _cryptoService.SetUserPrivateKeyAsync(newProtectedPrivateKey.EncryptedString);
+
+                // Set private key only for new JIT provisioned users in MP encryption orgs
+                // Existing TDE users will have private key set on sync or on login
+                if (keysRequest != null && ForceSetPasswordReason != ForcePasswordResetReason.TdeUserWithoutPasswordHasPasswordResetPermission)
+                {
+
+                    await _cryptoService.SetUserPrivateKeyAsync(keysRequest.EncryptedPrivateKey);
+                }
```

Comment:
I think both conditions overlap:
- `keysRequest` will be non-null if `ForceSetPasswordReason != ForcePasswordResetReason.TdeUserWithoutPasswordHasPasswordResetPermission`
- `keysRequest` will be null if `ForceSetPasswordReason == ForcePasswordResetReason.TdeUserWithoutPasswordHasPasswordResetPermission`
- `ForceSetPasswordReason` will not be `TdeUserWithoutPasswordHasPasswordResetPermission` if `keysRequest` is non-null
- `ForceSetPasswordReason` will be `TdeUserWithoutPasswordHasPasswordResetPermission` if `keysRequest` is null

Putting it in another way, `keysRequest` nullability directly depends on `ForceSetPasswordReason`.
So you can use any of the two, adding the second condition does nothing given that it will be evaluated the same as the first one. 
The only scenario I see it's worth is if in the near future we have another flow where the aforementioned conditions are not met the same way as it's today.

Reply:
Thanks for the detailed explanation i see what you mean!